components/delete_category_modal/delete_category_modal.tsx

```diff
@@ -32,81 +32,40 @@ export default class DeleteCategoryModal extends React.PureComponent<Props, Stat
         };
     }
 
-    handleCancel = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
-        event.preventDefault();
-        this.onHide();
-    }
-
-    handleConfirm = (event: React.MouseEvent<HTMLButtonElement>) => {
-        event.preventDefault();
+    handleConfirm = () => {
         this.props.actions.deleteCategory(this.props.category.id);
-        this.onHide();
-    }
-
-    onHide = () => {
-        this.setState({show: false}, this.props.onHide);
     }
 
     render() {
         return (
-            <Modal
-                dialogClassName='a11y__modal edit-category'
-                show={this.state.show}
-                onHide={this.onHide}
-                onExited={this.onHide}
-                enforceFocus={true}
-                restoreFocus={true}
-                role='dialog'
-                aria-labelledby='editCategoryModalLabel'
+            <GenericModal
+                onHide={this.props.onHide}
+                modalHeaderText={(
+                    <FormattedMessage
+                        id='delete_category_modal.deleteCategory'
+                        defaultMessage='Delete this category?'
+                    />
+                )}
+                handleCancel={() => {}}
```

Comment:
This is going to cause the `GenericModal` to re-render whenever this does. Can we just omit this prop? If we need to null check it in the component, we can also use the new `this.props.handleCancel?.()` syntax